src/primitive.rs

```diff
@@ -54,18 +56,20 @@ where
     fn to_le_bytes(self) -> Self::ByteRepr;
     fn from_le_bytes(repr: Self::ByteRepr) -> Self;
     fn leading_zeros(self) -> u32;
+    fn wrapping_sub(self, rhs: Self) -> Self;
 }
 
 pub(crate) trait PrimitiveSigned
 where
     Self: Copy,
     Self: TryFrom<Word>,
-    Self::Unsigned: PrimitiveUnsigned,
-    Self::Unsigned: TryFrom<Self>,
-    Self::Unsigned: TryInto<Self>,
+    Self: BitAnd<Output = Self>,
+    Self: Not<Output = Self>,
 {
-    type Unsigned;
+    type Unsigned: PrimitiveUnsigned + TryFrom<Self> + TryInto<Self>;
 
+    fn as_unsigned(self) -> Self::Unsigned;
```

Comment:
I believe this violates the standard naming conventions (https://rust-lang.github.io/api-guidelines/naming.html) -- `as_something` something is supposed to take reference and return reference.

So I suggest: `wrapping_to_unsigned` and `wrapping_from_unsigned`.